File: android/ac-rendering/src/main/kotlin/com/microsoft/adaptivecards/rendering/modifiers/SpacingModifier.kt
package com.microsoft.adaptivecards.rendering.modifiers

import androidx.compose.foundation.layout.padding
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.unit.dp
import com.microsoft.adaptivecards.core.models.Spacing
import com.microsoft.adaptivecards.hostconfig.LocalHostConfig

/**
 * Applies spacing based on Adaptive Card spacing property
 */
@Composable
fun Modifier.adaptiveSpacing(
    spacing: Spacing?,
    isFirst: Boolean = false
): Modifier {
    if (isFirst || spacing == null || spacing == Spacing.None) {
        return this
    }
    
    val hostConfig = LocalHostConfig.current
    val spacingValue = when (spacing) {
        Spacing.Small -> hostConfig.spacing.small
        Spacing.Default -> hostConfig.spacing.default
        Spacing.Medium -> hostConfig.spacing.medium
        Spacing.Large -> hostConfig.spacing.large
        Spacing.ExtraLarge -> hostConfig.spacing.extraLarge
        Spacing.Padding -> hostConfig.spacing.padding
        else -> 0
    }
    
    return this.padding(top = spacingValue.dp)
}

File: ios/Sources/ACRendering/Modifiers/SpacingModifier.swift
import SwiftUI
import ACCore

public extension View {
    /// Applies spacing based on the element's spacing property
    func spacing(_ spacing: Spacing?, hostConfig: HostConfig) -> some View {
        self.modifier(SpacingModifier(spacing: spacing, hostConfig: hostConfig))
    }
}

struct SpacingModifier: ViewModifier {
    let spacing: Spacing?
    let hostConfig: HostConfig

    func body(content: Content) -> some View {
        content.padding(.top, CGFloat(spacingValue))
    }

    private var spacingValue: Int {
        guard let spacing = spacing else { return 0 }

        switch spacing {
        case .none:
            return 0
        case .small:
            return hostConfig.spacing.small
        case .default:
            return hostConfig.spacing.default
        case .medium:
            return hostConfig.spacing.medium
        case .large:
            return hostConfig.spacing.large
        case .extraLarge:
            return hostConfig.spacing.extraLarge
        case .padding:
            return hostConfig.spacing.padding
        }
    }
}
